Dockerfile.ubuntu.18.04

```
@@ -5,24 +5,40 @@
 #
 # Pull base image
 FROM ubuntu:18.04
-Maintainer John.E.Malmberg <John.E.Malmberg@intel.com>
+Maintainer daos-stack <daos@daos.groups.io>
+
+# use same UID as host and default value of 1000 if not specified
+ARG UID=1000
 
 # Install basic tools
 RUN apt-get update && DEBIAN_FRONTEND=noninteractive apt-get install -y \
             autoconf bash curl debhelper dh-make dpkg-dev doxygen gcc \
-            git git-buildpackage locales make patch rpm wget
+            git git-buildpackage locales make patch pbuilder rpm wget
 
-# libfabric
-RUN apt-get install -y \
-            libibverbs-dev libpsm-infinipath1-dev librdmacm-dev
+# Add build user (to keep chrootbuild happy)
+ENV USER build
+RUN useradd -u $UID -ms /bin/bash $USER
 
-ENV UBUNTU_URL="https://launchpad.net/ubuntu/+archive/primary/+files"
-RUN curl -sS -f -L -O ${UBUNTU_URL}/libpsm2-2_10.3.58-1_amd64.deb ; \
-    curl -sS -f -L -O ${UBUNTU_URL}/libpsm2-dev_10.3.58-1_amd64.deb ; \
-    apt-get install -y \
-    ./libpsm2-2_10.3.58-1_amd64.deb ./libpsm2-dev_10.3.58-1_amd64.deb
+# need to run the build command as root, as it needs to chroot
+RUN if ! grep "^#includedir /etc/sudoers.d" /etc/sudoers; then              \
+        echo "#includedir /etc/sudoers.d" >> /etc/sudoers;                  \
+    fi;                                                                     \
+   echo "build ALL=(ALL) NOPASSWD: /usr/sbin/pbuilder" > /etc/sudoers.d/build; \
+   chmod 0440 /etc/sudoers.d/build;                                        \
+   visudo -c;                                                              \
+   sudo -l -U build
 
-# force an upgrade to get any newly built packages
-ARG CACHEBUST=1
-RUN apt-get update && DEBIAN_FRONTEND=noninteractive apt-get upgrade -y; \
-    apt-get autoremove -y
+ARG DAOS_PUB_KEY=""
+ARG DAOS_REPO=""
+ARG DAOS_REPO_SUP=""
+RUN if [ -n "${DAOS_PUB_KEY}" ]; then \
+       wget "${DAOS_REPO_SUP}/${DAOS_PUB_KEY}"; \
+       apt-key add "${DAOS_PUB_KEY}"; \
+       cp "${DAOS_PUB_KEY}" "/tmp/${DAOS_PUB_KEY}"; \
```

Do we need a copy of the key in this dir?

```suggestion
       mv "${DAOS_PUB_KEY}" "/tmp/${DAOS_PUB_KEY}"; \
```

But more importantly, this doesn't seem like something that should be hard-coded into the docker image.  If we had to change it we'd need to manually delete these docker images so that they are regenerated with a new key.